stl/inc/any

```diff
@@ -230,7 +230,10 @@ public:
     }
 
     void swap(any& _That) noexcept {
-        _That = _STD exchange(*this, _STD move(_That));
+        any _Old = _STD move(*this);
+        _Assign(_STD move(_That));

```

Comment:
This move-constructs `_Assign`'s value parameter (as an aside, I also don't like that pattern), which seems to be unnecessary. Self-swap correctness should already be ensured by capturing `any _Old` above.